src/plugins/intel_cpu/src/nodes/subgraph.cpp

```diff
@@ -499,7 +501,22 @@ void Snippet::SnippetJitExecutor::schedule_6d(const std::vector<MemoryPtr>& inMe
             int64_t indexes[] = {d0, d1, d2, d3, d4};
             jit_snippets_call_args call_args;
             update_ptrs(call_args, inMemPtrs, outMemPtrs);
+#ifdef __linux__
+            if (g_enable_snippets_err_detector) {
+                __sighandler_t signal_handler = [](int signal) {
+                    ov::intel_cpu::g_snippets_err_handler->local()->print_debug_info();
```

Comment:
I also tried to check how segfault_handler works: I compiled `ov_cpu_func_tests`, deliberately messed up the ptr increments in the LoopEnd emitter (I replaced [this](https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/emitters/x64/jit_snippets_emitters.cpp#L431) line with `h->add(data_ptr_regs[idx], ptr_increments[idx] * wa_increment * io_data_size[idx] + 3000000);`),  and run the tests with the following arguments:
`--gtest_filter=*smoke_Snippets_Eltwise*1.16.1.1*`

I expected to see segfault handler message, however, it didn't happen. It looks like `ov::intel_cpu::g_snippets_err_handler->local()` is nullptr. Did I do something wrong?

Reply:
Could you please try on latest commit by env var control to enable or disable by `export OV_CPU_SNIPPETS_ERR_DETECTOR=1` . It is caught and report as expect correctly on my machine.

Reply:
> It seems like `ov::intel_cpu::g_snippets_err_handler->local()` might be `nullptr` in some cases. Can we check it and not call `print_debug_info` for `nullptr` to avoid segfault inside `segfault_handler`?

done, thanks!

Reply:
> I also tried to check how segfault_handler works: I compiled `ov_cpu_func_tests`, deliberately messed up the ptr increments in the LoopEnd emitter (I replaced [this](https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/emitters/x64/jit_snippets_emitters.cpp#L431) line with `h->add(data_ptr_regs[idx], ptr_increments[idx] * wa_increment * io_data_size[idx] + 3000000);`), and run the tests with the following arguments: `--gtest_filter=*smoke_Snippets_Eltwise*1.16.1.1*`
> 
> I expected to see segfault handler message, however, it didn't happen. It looks like `ov::intel_cpu::g_snippets_err_handler->local()` is nullptr. Did I do something wrong?

Could you please try on latest commit by env var control to enable or disable by `export OV_CPU_SNIPPETS_ERR_DETECTOR=1` . It is caught and report as expect correctly on my machine.

Reply:
It works for me now. Thanks!